File: apps/my-service-admin/src/api/product/ProductCreateInput.ts
import { CategoryWhereUniqueInput } from "../category/CategoryWhereUniqueInput";

export type ProductCreateInput = {
  name?: string | null;
  description?: string | null;
  price?: number | null;
  sku?: string | null;
  category?: CategoryWhereUniqueInput | null;
};

File: apps/furniture-store/src/APIs/Product/Dtos/ProductCreateInput.cs
namespace FurnitureStore.APIs.Dtos;

public class ProductCreateInput
{
    public string? Id { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public string? Description { get; set; }

    public string? Name { get; set; }

    public double? Price { get; set; }

    public Category? Category { get; set; }

    public Supplier? Supplier { get; set; }
}
